File: part_5/content_assistant_web/src/services/blog_posts.ts
import axios from "axios"
import config from "../config";

type UUID = string;

export type BlogType = {
    id: number;
    guid: UUID;
    title: string;
    content: string;
    workflow_guid: UUID;
    created_at: Date;
    updated_at: Date;
    image_url: string;
}

export type Blogs = {
  total: number;
  blogs: [BlogType]
}


export const fetchBlogs = async (offset = 0, limit = 10) => {
    console.info('Fetching blogs...')
    return axios
      .get<Blogs>(`${config.CONTENT_SERVER_URL}/blogs`, {
        params: {
          offset,
          limit
        }
      })
      .then((r) => r.data)
  }

export const fetchBlog = async (blog_guid: string) => {
    console.info('Fetching blogs...')
    return axios
      .get<BlogType>(`${config.CONTENT_SERVER_URL}/blog`, {
        params: {
          blog_guid
        }
      })
      .then(r => r.data)
  }

File: part_5/content_assistant_server/src/content/schemas/blog_posts.py
import uuid
from typing import Optional
from pydantic import BaseModel
from datetime import datetime

class BlogCreate(BaseModel):
    title: str
    content: str
    workflow_guid: uuid.UUID
    
class BlogUpdate(BaseModel):
    id: int
    title: Optional[str] = None
    content: Optional[str] = None
    workflow_guid: Optional[uuid.UUID] = None
    image_guid: Optional[uuid.UUID] = None
    
class Blog(BaseModel):
    id: int
    guid: uuid.UUID
    title: str
    content: str
    workflow_guid: uuid.UUID
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True
        
class BlogList(BaseModel):
    total: int
    blogs: list[Blog]
    
    
class BlogWithImage(Blog):
    image_url: Optional[str] = None